src/02-common-principles.md

```diff
@@ -410,9 +409,3 @@ participants.tsv
 dataset_description.json
 CHANGES
 ```
-
-Additional files and folders containing raw data may be added as needed for
-special cases. They should be named using all lowercase with a name that
```

Comment:
@effigies - are you saying that this is no longer allowed?  or is there a different principled mechanism to follow when these special cases arise? for example, i may have motion corrected and non-motion corrected runs from the scanner. or i may have a file describing the prescribed gradient directions, as opposed to the recorded gradient directions. or say as heudiconv generates a grouping structure of the dicoms associated with each of the converted files.

Reply:
This paragraph appears to be suggesting that non-prescribed data types may be added ad libitum to a BIDS dataset, which is not consistent with anything else in the document, and will in fact put people in conflict with the validator. I'm not sure how you're interpreting this, though.

Reply:
> This paragraph appears to be suggesting that non-prescribed data types may be added ad libitum to a BIDS dataset, 

indeed i believe that was the intent. conform everything to bids, and then if it does not fit the current standard, this is how you can add the remaining pieces. the calibration example was an early precursor, but many examples exist.

i think the appropriate replacement text would be to say how non-prescribed data should be added. if everything needs to go through a formal: `can i add this to a bids dataset request`, then that should be clarified instead of just removing this?

so the question really is about the procedure when non-prescribed data arises (for which use cases are many).

Reply:
IMHO any non-prescribed data can be added to a BIDS dataset, as long as it is also added to the `.bidsignore` file. The point of it being non-prescribed is that the standard should not prescribe how to add it.

Reply:
> as long as it is also added to the .bidsignore file. 

isn't that a prescription? i think it's completely fine for a file-layout specification to indicate how something should be added that does not have a place in the current spec.

Reply:
In practice,I don't see why the standard should prescribe the _format_ of the filenames since any name is fine as long as it is in bidsignore. E.g., I am allowed to add `What-EvER-F1le-Name&+.Xtencion` I like.  It now states `Naming of files ... _should_ be lowercase_ ... and ...  _should_ follow the same scheme as ...`.  The _should_ in that sentence is confusing w.r.t. the _SHOULD_ as defined in https://www.ietf.org/rfc/rfc2119.txt. I don't think that the intention of the specification is as strict as implied by the RFC.

But more strictly: this relates to https://github.com/bids-standard/bids-specification/issues/131. I think it would be good if `.bidsignore` were standardized and documented in the standard. Without being able to refer to `.bidsignore` in the spec, any non-standard file currently violates the spec (but may pass the particular validator implementation, although python and javascript implementations might disagree). That is why I think that the spec (at this moment) should not specify that non-standard files can be added.

Reply:
Actually, I think the answer to this issue was sitting in the standard the whole time: in [src/02-common-principles.md](https://github.com/bids-standard/bids-specification/blob/master/src/02-common-principles.md) we have:

> This standard aspires to describe a majority of datasets, but acknowledges that there will be cases that do not fit. In such cases one can include additional files and subfolders to the existing folder structure following common sense. For example one may want to include eye tracking data in a vendor specific format that is not covered by this standard. The most sensible place to put it is next to the continuous recording file with the same naming scheme but different extensions. The solutions will change from case to case and publicly available datasets will be reviewed to include common data types in the future releases of the BIDS spec.

So... to me, that's clear:  BIDS is open to additional directories or files that don't conflict with the existing standard.   All this PR should do is perhaps make a revision to say 

1. Make additions with care, as future versions of BIDS may canonicalise a directory name you've used and change its meaning
2. Consult the details of your favorite validator which probably has a mechanism to suppress warnings of files outside the standard (or, we canonicalise `.bidsignore`, though there was much discussion on this in #131)
3. Fix the example... (I can't figure out what `sub-control01_calibration_Xcalibration.nii.gz` is supposed to be illustrating)

Reply:
i think the example is not helpful, and should be changed.

Reply:
but my main concern was with removing the first statement. perhaps this section could be a prescription for how to make additions (with care). and pointed to by that earlier section.

Reply:
How would this change affect existing BIDS examples such as [this one](https://github.com/bids-standard/bids-examples/blob/master/ds000117/.bidsignore) and [this one](https://github.com/bids-standard/bids-examples/blob/master/ds000248/.bidsignore)? I think the best way to understand the impact of any change to the specification is to consult the example datasets :)

Reply:
@effigies what is your opinion on  Thomas' [comment](https://github.com/bids-standard/bids-specification/pull/264/files#r300378768)? It seems like a compromise for now and we can get back to this after we have concluded the big discussion on the `.bidsignore` file.